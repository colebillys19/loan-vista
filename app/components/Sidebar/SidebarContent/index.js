/**
 * SidebarContent
 * @description ...
 */

import React, { Fragment } from 'react';
import T from 'prop-types';

import SidebarHeader from 'components/SidebarHeader';
import SidebarSummary from 'components/SidebarSummary';

const SidebarContent = ({
  pathname,
  sidebarHeaderData,
  sidebarSummaryData: {
    callsSummaryData,
    loanSummaryData,
    paymentSummaryData,
    serviceSummaryData,
  },
}) => (
  <Fragment>
    <SidebarHeader pathname={pathname} sidebarHeaderData={sidebarHeaderData} />
    <SidebarSummary data={loanSummaryData} iconName="bullet" title="Loan" />
    <SidebarSummary
      data={paymentSummaryData}
      iconName="bullet"
      title="Payment"
    />
    <SidebarSummary data={callsSummaryData} iconName="bullet" title="Calls" />
    <SidebarSummary
      data={serviceSummaryData}
      iconName="bullet"
      title="Service"
    />
  </Fragment>
);

SidebarContent.propTypes = {
  pathname: T.string.isRequired,
  sidebarHeaderData: T.object.isRequired,
  sidebarSummaryData: T.shape({
    callsSummaryData: T.array,
    loanSummaryData: T.array,
    paymentSummaryData: T.array,
    serviceSummaryData: T.array,
  }).isRequired,
};

export default SidebarContent;
