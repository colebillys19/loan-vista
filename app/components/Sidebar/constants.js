export const ROUTES_DICT = {
  '/': 'Home',
  '/calls': 'Calls',
  '/emails': 'Emails',
  '/payments': 'Payments',
  '/misc': 'Misc',
};
