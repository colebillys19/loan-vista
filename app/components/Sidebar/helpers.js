const validRoutes = ['/', '/calls', '/correspondence', '/payments', '/misc'];

export const pathnameIsValid = (pathname) =>
  validRoutes.indexOf(pathname) !== -1;
