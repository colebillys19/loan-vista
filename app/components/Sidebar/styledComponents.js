import styled from 'styled-components';

export const SidebarWrapper = styled.aside`
  background-color: #c7cbee;
  display: inline-block;
  min-height: calc(100vh - 14.3rem);
  padding: 1rem;
  position: absolute;
  width: 36rem;
`;
