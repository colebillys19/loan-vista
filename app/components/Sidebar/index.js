/**
 * Sidebar
 * @description ...
 */

import React from 'react';
import T from 'prop-types';

import SidebarHeader from 'components/SidebarHeader';
import SidebarSummary from 'components/SidebarSummary';
import { SidebarContainer, SidebarContentContainer } from './styledComponents';

const Sidebar = ({
  pathname,
  sidebarHeaderData,
  sidebarSummariesData: {
    callsSummaryData,
    loanSummaryData,
    paymentSummaryData,
    serviceSummaryData,
  },
}) => (
  <SidebarContainer>
    <SidebarContentContainer>
      <SidebarHeader
        pathname={pathname}
        sidebarHeaderData={sidebarHeaderData}
      />
      <SidebarSummary data={loanSummaryData} iconName="bullet" title="Loan" />
      <SidebarSummary
        data={paymentSummaryData}
        iconName="bullet"
        title="Payment"
      />
      <SidebarSummary data={callsSummaryData} iconName="bullet" title="Calls" />
      <SidebarSummary
        data={serviceSummaryData}
        iconName="bullet"
        title="Service"
      />
    </SidebarContentContainer>
  </SidebarContainer>
);

Sidebar.propTypes = {
  pathname: T.string.isRequired,
  sidebarHeaderData: T.object.isRequired,
  sidebarSummariesData: T.shape({
    callsSummaryData: T.array,
    loanSummaryData: T.array,
    paymentSummaryData: T.array,
    serviceSummaryData: T.array,
  }).isRequired,
};

export default Sidebar;
