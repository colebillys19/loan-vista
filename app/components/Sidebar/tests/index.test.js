import React from 'react';
import { render } from 'react-testing-library';

import Sidebar from '../index';

const mockProps = {
  error: false,
  loading: false,
  loanNumber: '',
  pathname: '/',
  sidebarHeaderData: {
    address1: 'address1',
    address2: 'address2',
    health: 0,
    loanNumber: 'loanNumber',
    name: 'name',
  },
  sidebarSummariesData: {
    callsSummaryData: [{ label: 'label', value: 'value' }],
    health: 0,
    loanSummaryData: [{ label: 'label', value: 'value' }],
    paymentSummaryData: [{ label: 'label', value: 'value' }],
    serviceSummaryData: [{ label: 'label', value: 'value' }],
  },
};

const Component = <Sidebar {...mockProps} />;

describe('<Sidebar />', () => {
  it('Expect not to log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(Component);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Expect to render', () => {
    const { container } = render(Component);
    expect(container.firstChild).toBeDefined();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(Component);
    expect(firstChild).toMatchSnapshot();
  });
});
