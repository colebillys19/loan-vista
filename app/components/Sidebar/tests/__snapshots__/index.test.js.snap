// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> Should render and match the snapshot 1`] = `
<aside
  class="styledComponents__SidebarWrapper-sc-1es1u17-0 cririz"
>
  <div
    class="styledComponents__SidebarHeaderWrapper-avjjg6-1 lhvoSt"
  >
    <div
      class="styledComponents__IconWrapper-avjjg6-0 fOWgsI"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        data-name="construction crenelation building labor sign"
        focusable="false"
        height="512"
        id="construction_crenelation_building_labor_sign"
        style="height: 10rem; width: 10rem;"
        viewBox="0 0 64 64"
        width="512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m31.396 7.5-28.396 48.24v1.76h29 29v-1.76l-28.396-48.24z"
          fill="#4370ff"
        />
        <path
          d="m30 47.5h4v4h-4z"
        />
        <path
          d="m30 24.244h4v19.256h-4z"
        />
        <path
          d="m33.748 5.5h-3.496l-29.252 49.695v4.305h33v-4h-28.538l26.538-45.084 26.538 45.084h-20.538v4h25v-4.305z"
        />
      </svg>
    </div>
    <h1
      class="styledComponents__StyledH1-avjjg6-2 hcHndy"
    >
      Not Found
    </h1>
  </div>
</aside>
`;
